NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/ClientIpHeaderTelemetryInitializer.cs

```diff
@@ -65,6 +65,21 @@ public string HeaderValueSeparators
         /// <inheritdoc />
         protected override void OnInitializeTelemetry(HttpContext platformContext, RequestTelemetry requestTelemetry, ITelemetry telemetry)
         {
+            if (telemetry == null)
+            {
+                throw new ArgumentNullException(nameof(telemetry));
+            }
+
+            if (platformContext == null)
+            {
+                throw new ArgumentNullException(nameof(platformContext));
```

Comment:
This now throws if even if requestTelemetry.Context.Location.Ip is not empty. We need to check if upstream code is Ok with this.

Reply:
moved into if